packages/ra-core/src/dataProvider/addOfflineSupportToQueryClient.ts

```diff
@@ -0,0 +1,102 @@
+ *
+ * @example <caption>Adding offline support for the default mutations</caption>
+ * // in src/App.tsx
+ * import { Admin, Resource, addOfflineSupportToQueryClient, reactAdminMutations } from 'react-admin';
+ * import { PersistQueryClientProvider } from '@tanstack/react-query-persist-client';
+ * import { createSyncStoragePersister } from '@tanstack/query-sync-storage-persister';
+ * import { queryClient } from './queryClient';
+ * import { dataProvider } from './dataProvider';
+ * import { posts } from './posts';
+ * import { comments } from './comments';
+ *
+ * const localStoragePersister = createSyncStoragePersister({
+ *     storage: window.localStorage,
+ * });
+ *
+ * addOfflineSupportToQueryClient({
+ *    queryClient,
+ *   dataProvider,
+ *   resources: ['posts', 'comments'],
+ *   mutations: [...reactAdminMutations, 'myCustomMutation'],
+ * });
+ *
+ * const App = () => (
+ *     <PersistQueryClientProvider
+ *         client={queryClient}
+ *         persistOptions={{ persister: localStoragePersister }}
+ *         onSuccess={() => {
+ *             // resume mutations after initial restore from localStorage was successful
+ *             queryClient.resumePausedMutations();
+ *         }}
+ *     >
+ *         <Admin queryClient={queryClient} dataProvider={dataProvider}>
+ *             <Resource name="posts" {...posts} />
+ *             <Resource name="comments" {...comments} />
+ *         </Admin>
+ *     </PersistQueryClientProvider>
+ * );
+ *
+ * @example <caption>Adding offline support with custom mutations</caption>
+ * // in src/App.tsx
+ * import { addOfflineSupportToQueryClient } from 'react-admin';
+ * import { PersistQueryClientProvider } from '@tanstack/react-query-persist-client';
+ * import { createSyncStoragePersister } from '@tanstack/query-sync-storage-persister';
+ * import { queryClient } from './queryClient';
+ * import { dataProvider } from './dataProvider';
+ * import { posts } from './posts';
+ * import { comments } from './comments';
+ *
+ * const localStoragePersister = createSyncStoragePersister({
+ *     storage: window.localStorage,
+ * });
+ *
+ * addOfflineSupportToQueryClient({
+ *    queryClient,
+ *   dataProvider,
+ *   resources: ['posts', 'comments'],
+ * });
+ *
+ * const App = () => (
+ *     <PersistQueryClientProvider
+ *         client={queryClient}
+ *         persistOptions={{ persister: localStoragePersister }}
+ *         onSuccess={() => {
+ *             // resume mutations after initial restore from localStorage was successful
+ *             queryClient.resumePausedMutations();
+ *         }}
+ *     >
+ *         <Admin queryClient={queryClient} dataProvider={dataProvider}>
+ *             <Resource name="posts" {...posts} />
+ *             <Resource name="comments" {...comments} />
+ *         </Admin>
+ *     </PersistQueryClientProvider>
+ * );
+ */
+export const addOfflineSupportToQueryClient = ({
```

Comment:
As explained earlier, can't we create a new Query Client instead? or at least return the modified one.